### ISSUE TITLE
Signed Exchanges: understand AMP runtime's needs for serving origin identity

### ISSUE THREAD

**Issue by Gregable** [ampproject/amphtml#20359]
@ampproject/wg-ui-and-a11y 

With Signed Exchanges, AMP documents will be operating on the publisher origin, even if the document is served from the AMP Cache. By default, the AMP runtime will not be able to determine if the document was served by the AMP cache or the Publisher.

The open question is if the runtime makes any decisions based on the origin being `*.cdn.ampproject.org`, etc. A hypothetical example could be rewriting images in `amp-bind` with a `cdn.ampproject.org` cache URL. 

If the AMP runtime does not make any such decisions, this issue can simply be closed. If it does, signed exchanges can provide metadata in the document which the runtime can use to make that decision. We should determine what the syntax of this metadata should be.

@honeybadgerdontcare for the syntax of the `<html amp transformed=>` attribute which may be sufficient.

If not, we can also do something like:

```<meta name="amp-cache-origin" content="cdn.ampproject.org">```

**Comment by honeybadgerdontcare** [comment]
The AMP Packager transformer [transformerIdentifier.go](https://github.com/ampproject/amppackager/blob/master/transformer/transformers/transformedidentifier.go) will add the attribute `transformed` with value `google;v=1` to the `<html>` tag. The value is validated by the AMP Validator.

This follows the form of the [AMP-Cache-Transform](https://github.com/ampproject/amphtml/blob/master/spec/amp-cache-transform.md) header. It explicitly states the CDN performing the transform and the transformer version used.

E.g. `<html amp transformed="google;v=1">`

**Comment by alanorozco** [comment]
My 2c: [There is a manageable number of hard origin checks throughout the runtime.](https://github.com/ampproject/amphtml/search?q=isProxyOrigin&unscoped_q=isProxyOrigin) In many cases this is just a proxy signal for whether the document is being loaded in a Google viewer context, so the proposed API should be sufficient.

**Comment by Gregable** [comment]
Do you have thoughts on which API would be preferred? Is `transformed=google` sufficient, or should the cache provide the actual origin?

**Comment by choumx** [comment]
Currently there are several configurable data (including origin) that other caches can theoretically override: https://github.com/ampproject/amphtml/blob/6f63baff846974ec9a33889221f8a6c5e3f2c076/src/config.js#L24-L47

There are also a few places where cdn.ampproject.org logic is hardcoded e.g. [parsing](https://github.com/ampproject/amphtml/blob/6f63baff846974ec9a33889221f8a6c5e3f2c076/src/url.js#L502) and [generating](https://github.com/ampproject/amphtml/blob/6f63baff846974ec9a33889221f8a6c5e3f2c076/src/purifier.js#L720-L724) cache URLs.

Providing the cache origin would be nice (obviates `cdnUrl` above) but insufficient to decouple runtime from cache. If that's not necessary for SxG then `transformed=google` should suffice.

**Comment by Gregable** [comment]
I see. Well, presumably other caches could also provide their own javascript URL on the documents as well.

I then propose that we stick with `transformed=google...` for now until we have a real use case to consider. Some of these decisions depend on the number of caches we'd need to support. For example, if there are only 2 or 3, then it's reasonable to encode these parameters into every page. If there are hundreds, then we would not want to bloat the javascript file size with the configs.

Wherever we have the hard origin checks, we should replace that with a call that checks against `transformed=google...`. Note that the AMP cache sets this already, so it should be sufficient to _only_ consider this html attribute.

**Comment by honeybadgerdontcare** [comment]
It may be worth noting that eventually the concept of transformed AMP will come to pages served at origin. The actual attribute value hasn't been determined yet (e.g. `transformed=self` or `transformed=amp-toolkit`). When these are seen by an AMP cache they can choose to ingest them and apply their own transforms on top of them causing the final attribute value to be that of the cache. In Google's case `transformed=self` would become `transformed=google;v=#`.

Also, an AMP page served from origin could declare `transformed=google;v=#`. So whatever the AMP runtime does in the cases where it thinks it's from an AMP cache, it'll need to be something that doesn't fully break the page.

**Comment by Gregable** [comment]
We discussed this for ~5 minutes offline and concluded that for now `transformd=google...` is a sufficient marker for AMP runtime.

The other suggestion that was raised is that other caches could implement these modifications in the future by mandating an alternative source for `v0.js` which makes the necessary changes to the runtime config.

Closing this issue. @jridgewell for filing any issues that for runtime changes that may be needed.
